abi/nested/template.go

```diff
@@ -0,0 +1,51 @@
+package nested
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"strings"
+	"text/template"
+
+	"github.com/indexsupply/x/abi"
+)
+
+const ct = `
+{{with .Event}}
+type {{.Name}} struct {
+	{{range .Inputs}}
+		{{geninput .}}
```

Comment:
I didn't end up using `define` since having a custom go function appeared more readable to me

Reply:
```
{{ define "x" }}
{{ if gt (len .Comp) 0 }}
{{ .Name }} struct { 
    {{ range .Comp }}
    {{ template "x" . }}
    {{ end }}
}
{{ else }}
{{ .Name }} {{ .Gtyp }}
{{ end }}
{{ end }}

type {{ template "x" . }}
```

Reply:
I kind of like solving it in the template for a couple of reasons:
1) By using another function we spread the template building code into 2 places with effectively two different semantic sets: go/sprintf AND text/emplate. I think it's simpler to either do everything with sprintf or everything with template.
2) The recursive template solution is no less readable then the go function